log_parser/event_detector.py

```diff
@@ -0,0 +1,62 @@
+from typing import List
+
+from log_parser.log import Log
+from log_parser.event_pattern import EventPattern
+from log_parser.detected_event import DetectedEvent
+from log_parser.log_sequence_matcher import LogSequenceMatcher
+
+
+class EventDetector:
+    """
+    LogParser class to parse log entries and detect events based on defined patterns.
+    """
+
+    def __init__(self, event_patterns: List[EventPattern]):
+        """Initialize the LogParser with a list of event patterns."""
+        self._event_patterns = event_patterns
+        self._matchers = []
+
+    def detect_events(self, logs: List[Log]) -> List[DetectedEvent]:
+        """
+        Detect events in the provided logs based on the defined event patterns.
+
+        Args:
+            logs (List[Log]): A list of log entries
+        """
+        detected_events = []
+        for log in logs:
+            # Check if the log message matches the start of any event pattern
+            # If it does, create a new LogSequenceHandler for that pattern
+            for pattern in self._event_patterns:
+                if pattern.matches_pattern_at_index(log.message):
+                    self._matchers.append(LogSequenceMatcher(pattern, log.timestamp))
+            # Process the current log entry against all active handlers
+            detected_events += self._process_handlers(log)
+        return detected_events
+
+    def _process_handlers(self, log: Log) -> List[DetectedEvent]:
```

Comment:
Rename, now it tries to match the given log to all active log sequence matchers.
Change the documentation accordingly